app/controllers/likes_controller.rb

```diff
@@ -0,0 +1,21 @@
+class LikesController < ApplicationController
+  before_action :find_post
+  def create
+    if already_liked?
+      flash[:notice] = "You can't like more than once"
+    else
+      @post.likes.create(user_id: current_user.id)
+    end
+    redirect_to URI(request.referer).path
+  end
+
+  private
+  def find_post
+    @post = Post.find(params[:post_id])
+  end
+
+  def already_liked?
```

Comment:
Nice, I like how this was refactored into its own method. 
Perhaps also make it a private method

Reply:
I just tried to put it in Private but the web broke 🙈